### PR TITLE
Allow attributes be readonly in backend

### PR DESCRIPTION
**Pull request by pascalheidmann-bedarf**
### 1. Why is this change necessary?
Sometimes you want to add a attribute that should not be editable for your users but still visible in backend. This allows to make that attribute be readonly.

### 2. What does this change do, exactly?
Adding a new attribute configuration option that marks that attribute as readonly. This setting is then used to disable the input field in extjs. Therefore changes by code are of course still possible

### 5. Which documentation changes (if any) need to be made because of this PR?
New option "readonly" for attribute cru(d).

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Bundle/AttributeBundle/Service/ConfigurationStruct.php [hunk 1]
```diff
@@ -227,6 +231,22 @@ public function setDisplayInBackend($displayInBackend)
         $this->displayInBackend = $displayInBackend;
     }
 
+    /**
+     * @return bool
+     */
+    public function readonly()
```
Comment (shyim) [inline engine/Shopware/Bundle/AttributeBundle/Service/ConfigurationStruct.php]: Should the name of the function not isReadOnly?

Comment (pascalheidmann-bedarf) [reply engine/Shopware/Bundle/AttributeBundle/Service/ConfigurationStruct.php]: You are right. Copy/pasting displayInBackend properties ;)

**Comment by buddhaCode** [comment]
Love your PR, @pascalheidman-bedarf. I had the same idea and have the same thing in one of my stashes :)

**Comment by FiveDigital** [comment]
Perhaps this can be merged into 5.5 if @pascalheidman-bedarf changes the target branch?

**Comment by pascalheidmann-bedarf** [comment]
@shopwareBot have fixed the row editor (but forgot to sync latest changes that use `FQN::class` instead of classnames directly as strings).
Will look into the HTML editor later but it seems this is a shortcoming in the `Shopware.form.field.TinyMCE` class I have no clue about yet ;)

File: engine/Shopware/Components/MultiEdit/Resource/Product/DqlHelper.php [hunk 2]
```diff
@@ -1027,4 +1035,23 @@ private function showColumnInGrid($column)
 
         return $show;
     }
+
+    private function loadReadOnlyAttribute(array $fields) {
```
Comment (shyim) [inline engine/Shopware/Components/MultiEdit/Resource/Product/DqlHelper.php:1039]: PHP Docs and the code style is wrong :wink:

**Comment by shyim** [comment]
Tinymce can be easily fixed with a if condition here https://github.com/shopware/shopware/blob/c75f8af607857acafe07c0374bed59c37363828b/themes/Backend/ExtJs/backend/base/attribute/field_handler/Shopware.attribute.HtmlFieldHandler.js#L39

if attribute has disabled true set field.readOnly = true; (https://github.com/shopware/shopware/blob/5.5/themes/Backend/ExtJs/backend/base/component/Shopware.form.field.TinyMCE.js#L333)

**Pull request closed without merging**